src/OrchardCore/OrchardCore.ContentManagement/DefaultContentManager.cs

```diff
@@ -438,13 +438,18 @@ public async Task CreateAsync(ContentItem contentItem, VersionOptions options)
                 contentItem.Published = false;
             }
 
+            if (options.IsPublished)
+            {
+                contentItem.Published = true;
+            }
```

Comment:
I haven't run the code but this is the fix I was expecting.

On line 457 it seems like it would be safer to do `if (contentItem.Published)` instead of `if (options.IsPublished)` when deciding to fire the publish events.

Reply:
I'm not sure why we assumed that the ContentItem would be set to Published = true here. I think the method was used to publish all the time at first. And then we decided that we would need to create a draft and then publish it. Need to see before merging if the logic make sense in all use cases we have.

Reply:
Else, not sure why we would need to have a VersionOptions passed in these methods. There's probably a reason and I need to find why first.

Reply:
I think the idea is that you could alter the ContentItem in any of the Events fired. So, if we pass a VersionOptions.Published then we retain the main intent and then don't need to look if the ContentItem.Published = true. Though, the issue here is that if it's not set to Published it won't create a Published content item. Then, I think you could have set your ContentItem.Published = true before calling UpdateValidateAndCreateAsync method. Just thinking out loud but need to validate.